### PR TITLE
Align RDD.takeSample with PySpark RDDs

### LINKED ISSUE #90: RDD.takeSample does not support withReplacement=False
Hello,

`RDD.takeSample` input parameters are currently in pysparkling as such:
```python
def takeSample(self, n):
   ...
```

Whereas PySpark defines it like that:
```python
def takeSample(self, withReplacement, num, seed=None):
   ...
```
Besides, in the case of a RDD with 5 items, pysparkling RDD.takeSample(n=200) will return 200 items no matter what, and there is no guarantee that all of the 5 items are in the 200.

In PySpark, there is the guarantee and it returns 200 items only if withReplacement is True. 5 items otherwise.

As the PySpark implementation is 100% in Python, I'd suggest merging their code in pysparkling in order to have the same features and behaviour.

### PR DESCRIPTION
**Pull request by tools4origins**
Fix #90

### CODE REVIEW

File: pysparkling/rdd.py [hunk 1]
```diff
@@ -1844,7 +1872,7 @@ def __init__(self, prev, sampler, preservesPartitioning=False,
         RDD.__init__(self, prev.partitions(), prev.context)
 
         if seed is None:
-            seed = random.randint(0, maxint)
+            seed = random.randint(0, 2**31)
```
Comment (tools4origins) [inline pysparkling/rdd.py]: numpy (that is used after) limits seed to (2**32)-1

**Comment by svenkreiss** [comment]
Hi @tools4origins , thanks for sending another PR. Very much appreciated.

You mentioned in the issue something about copying code. I don't know whether you have done that here now, but I want to make sure everything is done according to the applicable licenses. Is that the case? Thanks.

**Comment by tools4origins** [comment]
Most of the new code is coming from PySpark counterparts,

Do you want to add a notice similar to what was added in https://github.com/svenkreiss/pysparkling/pull/29/commits/69c1d7585044d7951c46240a5bd9866c70ee5dcf ?

**Comment by tools4origins** [comment]
Some tests used to fail because there are parsing the LICENSE file which was modified,

I've updated the PR with some other style fixes requested by flake8,

It builds now :)

**Pull request merged**